From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kirill Yatsenko <kiriyatsenko@gmail.com>
Date: Thu, 03 Apr 2026 13:00:00 +0200
Subject: dts: Add device tree for decktrix custom PCB

The decktrix board is based on STM32MP157F with the same RAM as the
Discovery Kit but different pinout and peripherals:

- Jadard JD9365TN display (different reset/backlight GPIOs)
- Jadard touch controller on I2C1 (different IRQ/reset GPIOs)
- STPMIC1 PMIC on I2C4 at PD12/PD13 instead of PZ4/PZ5
- LDO1 at 2.8V for display AVDD (not 1.8V for audio)
- LDO6 at 1.8V for display IOVCC (not 1.2V for HDMI)
- Display powered by always-on PMIC regulators (no GPIO enable)
- WiFi/BT disabled (u-blox MAYA-W166 uses SDMMC3/UART8, TBD)

Signed-off-by: Kirill Yatsenko <kiriyatsenko@gmail.com>
---
 arch/arm/boot/dts/st/Makefile          |   1 +
 arch/arm/boot/dts/st/decktrix-v1.dts   | 118 ++++++++++++++++++++++++
 2 files changed, 119 insertions(+)
 create mode 100644 arch/arm/boot/dts/st/decktrix-v1.dts

diff --git a/arch/arm/boot/dts/st/Makefile b/arch/arm/boot/dts/st/Makefile
index 75346d084d2d..8b3d8d4c9e1f 100644
--- a/arch/arm/boot/dts/st/Makefile
+++ b/arch/arm/boot/dts/st/Makefile
@@ -62,6 +62,7 @@ dtb-$(CONFIG_ARCH_STM32) += \
 	stm32mp157c-dk2.dtb \
 	stm32mp157c-dk2-scmi.dtb \
 	stm32mp157c-dk2-jadard.dtb \
+	decktrix-v1.dtb \
 	stm32mp157c-ed1.dtb \
 	stm32mp157c-ed1-scmi.dtb \
 	stm32mp157c-emsbc-argon.dtb \
diff --git a/arch/arm/boot/dts/st/decktrix-v1.dts b/arch/arm/boot/dts/st/decktrix-v1.dts
new file mode 100644
index 000000000000..000000000000
--- /dev/null
+++ b/arch/arm/boot/dts/st/decktrix-v1.dts
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
+/*
+ * Copyright (C) Decktrix Lab
+ * Author: Kirill Yatsenko <kiriyatsenko@gmail.com>
+ *
+ * Decktrix v1 custom PCB based on STM32MP157F.
+ * Same SoC and RAM as STM32MP157F-DK2 but with custom pinout:
+ *   - Jadard JD9365TN display (different reset/backlight GPIOs)
+ *   - Jadard touch on I2C1 (different IRQ/reset GPIOs)
+ *   - STPMIC1 PMIC on I2C4 at PD12/PD13 (not PZ4/PZ5)
+ *   - LDO1 at 2.8V for display AVDD (not 1.8V for audio)
+ *   - LDO6 at 1.8V for display IOVCC (not 1.2V for HDMI)
+ *   - WiFi/BT disabled (u-blox MAYA-W166 uses SDMMC3/UART8, TBD)
+ */
+
+/dts-v1/;
+
+#include "stm32mp157.dtsi"
+#include "stm32mp15xc.dtsi"
+#include "stm32mp15-pinctrl.dtsi"
+#include "stm32mp15xxac-pinctrl.dtsi"
+#include "stm32mp15xx-dkx.dtsi"
+
+/ {
+	model = "Decktrix v1";
+	compatible = "decktrix,v1", "st,stm32mp157";
+
+	aliases {
+		ethernet0 = &ethernet0;
+	};
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+};
+
+&cryp1 {
+	status = "okay";
+};
+
+&dsi {
+	#address-cells = <1>;
+	#size-cells = <0>;
+	status = "okay";
+
+	/* Jadard JD9365TN 480x1080 MIPI DSI panel */
+	panel@0 {
+		compatible = "shenzen,z34014p30365ty1", "jadard,jd9365tn";
+		reg = <0>;
+		reset-gpios     = <&gpiod 11 GPIO_ACTIVE_HIGH>;
+		backlight-gpios = <&gpiof 6  GPIO_ACTIVE_HIGH>;
+		status = "okay";
+
+		port {
+			panel_in: endpoint {
+				remote-endpoint = <&dsi_out>;
+			};
+		};
+	};
+};
+
+&dsi_in {
+	remote-endpoint = <&ltdc_ep1_out>;
+};
+
+&dsi_out {
+	remote-endpoint = <&panel_in>;
+};
+
+/* PMIC LDO1: 2.8V for display AVDD (2V8 rail on decktrix PCB) */
+&v1v8_audio {
+	regulator-name = "ldo1";
+	regulator-min-microvolt = <2800000>;
+	regulator-max-microvolt = <2800000>;
+};
+
+/* PMIC LDO6: 1.8V for display IOVCC (1V8 rail on decktrix PCB) */
+&v1v2_hdmi {
+	regulator-name = "ldo6";
+	regulator-min-microvolt = <1800000>;
+	regulator-max-microvolt = <1800000>;
+};
+
+&i2c1 {
+	jadard@68 {
+		compatible = "jadard,jdcommon";
+		reg = <0x68>;
+		interrupts-extended = <&gpiog 7 IRQ_TYPE_EDGE_FALLING>;
+		jadard,panel-sense-nums = <30 48>;
+		jadard,panel-coords = <0 480 0 1080>;
+		jadard,panel-max-points = <10>;
+		jadard,int-is-edge = <1>;
+		jadard,irq-gpio = <&gpiog 7 GPIO_ACTIVE_HIGH>;
+		jadard,rst-gpio = <&gpioe 10 GPIO_ACTIVE_HIGH>;
+		status = "okay";
+	};
+};
+
+&ltdc {
+	status = "okay";
+
+	port {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		ltdc_ep1_out: endpoint@1 {
+			reg = <1>;
+			remote-endpoint = <&dsi_in>;
+		};
+	};
+};
+
+/* WiFi disabled - u-blox MAYA-W166 uses SDMMC3, driver TBD */
+&sdmmc2 {
+	status = "disabled";
+};
+
+/* BT disabled - u-blox MAYA-W166 uses UART8, driver TBD */
+&uart7 {
+	status = "disabled";
+};
-- 
2.43.0
